lib/pools.js

```diff
@@ -0,0 +1,43 @@
+import mssql from "mssql";
+
+export const pools = new Map();
+
+export default {
+  get: (name, config, driver) => {
+    if (!pools.has(name)) {
+      if (!config) {
+        throw new Error("Database configuration required");
+      }
+
+      if (!driver) {
+        throw new Error("Driver is required");
+      }
+
+      let pool;
+      switch (driver) {
+        case "mssql":
```

Comment:
I think you'll want your pools to be completely driver agnostic. I'd recommend to rely on dependency injection and pass in a driver / callback / instance instead of implementing the instantiation here.

Reply:
If certain connectors handle their connections differently, then you'll probably want to hide that implementation detail behind a common API.